automation/new_tests.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if [[ "$#" -eq 0 ]] || [[ -z "$1" ]]; then
+    TARGET_COMMIT=$(git log -1 --format=%H --merges)
+else
+    TARGET_COMMIT="$1"
+fi
+git diff --name-only "${TARGET_COMMIT}".. -- tests/ \
```

Comment:
would you mind adding a comment describing what this block does?
decoding the regex and all the pipes is not that straight forward... :)

Reply:
Makes sense.